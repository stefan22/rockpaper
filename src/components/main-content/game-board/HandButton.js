import React from 'react';
import Paper from '@material-ui/core/Paper';
import Typography from '@material-ui/core/Typography';

const HandButton = (props) => {

  return (
    <div className='col-4'>
      <Paper className='game-hand'>
        <img
          className='rps-button'
          onClick={() => props.onPlayerHandPick(props.choice)}
          src={props.imgsrc} alt={props.imgsrc}
        />
        <Typography className='game-name' gutterBottom variant="h4" component="h4">
          {props.hand}
        </Typography>

      </Paper>
    </div>
  );

};


export default HandButton;




