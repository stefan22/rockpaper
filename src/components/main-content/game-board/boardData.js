export const boardData = () => {
  const board = {
    nGames:'GAMES',
    nWins: 'WINS',
    nLosses:'LOSSES',
    nDraws: 'DRAWS',
    nTurn: 'TURN',
  };

  return board;
};






