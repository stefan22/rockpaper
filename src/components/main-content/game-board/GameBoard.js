import React, {Component} from 'react';
import Table from '@material-ui/core/Table';
import TableBody from '@material-ui/core/TableBody';
import TableHead from '@material-ui/core/TableHead';
import TableCell from '@material-ui/core/TableCell';
import TableRow from '@material-ui/core/TableRow';
import Paper from '@material-ui/core/Paper';
import {boardData} from './boardData';
import '../../../scss/markets.scss';
import '../../../scss/boardgame.scss';

class GameBoard extends Component {
  constructor(props) {
    super(props);
    this.state ={
      board:[],
    };
    this.getBoard = this.getBoard.bind(this);
  }

  componentDidMount() {
    this.getBoard();
  }

  getBoard() {
    let board = boardData();
    return this.setState({
      board,
    });
  }

  render() {
    const {nGames,nWins,nLosses,nDraws,nTurn} = this.state.board;
    let {games,turn,wins,draws,losses} = this.props;

    return (
      <Paper className='paper-root'>

        <Table id='rpc-table'>
          <TableHead>
            <TableRow>
              <TableCell className='ngames' align='left'>{nGames}</TableCell>
              <TableCell className='wins' align='left'>{nWins}</TableCell>
              <TableCell className='losses' align='left'>{nLosses}</TableCell>
              <TableCell className='draws' align='left'>{nDraws}</TableCell>
              <TableCell className='turn' align='left'>{nTurn}</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            <TableRow
              className='score-row'>
              <TableCell className='games' align='left'>
                {games}
              </TableCell>
              <TableCell className='wins' align='left'>
                {wins}
              </TableCell>
              <TableCell className='losses' align='left'>
                {losses}
              </TableCell>
              <TableCell className='draws' align='left'>
                {draws}
              </TableCell>
              <TableCell className='isturn' align='left'>
                {turn}
              </TableCell>
            </TableRow>
          </TableBody>
        </Table>
      </Paper>
    );
  }
}



export default GameBoard;








