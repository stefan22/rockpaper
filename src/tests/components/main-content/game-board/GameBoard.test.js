import React from 'react';
import {shallow} from 'enzyme';
import GameBoard from  '../../../../components/main-content/game-board/GameBoard';
import {boardData} from '../../../../components/main-content/game-board/boardData';

describe('GameBoard component', () => {
  it('should have "Game/Wins/Losses/Draws and Turn as column titles"', () => {
    const wrapper = shallow(<GameBoard boardData={boardData} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should have table id of "rpc-table"', () => {
    const wrapper = shallow(<GameBoard />);
    expect(wrapper.find('#rpc-table').length).toBe(1);
  });



});


