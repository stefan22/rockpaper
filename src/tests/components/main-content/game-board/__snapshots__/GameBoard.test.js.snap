// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GameBoard component should have "Game/Wins/Losses/Draws and Turn as column titles" 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GameBoard
    boardData={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": <WithStyles(Table)
        id="rpc-table"
      >
        <WithStyles(TableHead)>
          <WithStyles(TableRow)>
            <WithStyles(TableCell)
              align="left"
              className="ngames"
            >
              GAMES
            </WithStyles(TableCell)>
            <WithStyles(TableCell)
              align="left"
              className="wins"
            >
              WINS
            </WithStyles(TableCell)>
            <WithStyles(TableCell)
              align="left"
              className="losses"
            >
              LOSSES
            </WithStyles(TableCell)>
            <WithStyles(TableCell)
              align="left"
              className="draws"
            >
              DRAWS
            </WithStyles(TableCell)>
            <WithStyles(TableCell)
              align="left"
              className="turn"
            >
              TURN
            </WithStyles(TableCell)>
          </WithStyles(TableRow)>
        </WithStyles(TableHead)>
        <WithStyles(TableBody)>
          <WithStyles(TableRow)
            className="score-row"
          >
            <WithStyles(TableCell)
              align="left"
              className="games"
            />
            <WithStyles(TableCell)
              align="left"
              className="wins"
            />
            <WithStyles(TableCell)
              align="left"
              className="losses"
            />
            <WithStyles(TableCell)
              align="left"
              className="draws"
            />
            <WithStyles(TableCell)
              align="left"
              className="isturn"
            />
          </WithStyles(TableRow)>
        </WithStyles(TableBody)>
      </WithStyles(Table)>,
      "className": "paper-root",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithStyles(TableHead)>
            <WithStyles(TableRow)>
              <WithStyles(TableCell)
                align="left"
                className="ngames"
              >
                GAMES
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="wins"
              >
                WINS
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="losses"
              >
                LOSSES
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="draws"
              >
                DRAWS
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="turn"
              >
                TURN
              </WithStyles(TableCell)>
            </WithStyles(TableRow)>
          </WithStyles(TableHead)>,
          <WithStyles(TableBody)>
            <WithStyles(TableRow)
              className="score-row"
            >
              <WithStyles(TableCell)
                align="left"
                className="games"
              />
              <WithStyles(TableCell)
                align="left"
                className="wins"
              />
              <WithStyles(TableCell)
                align="left"
                className="losses"
              />
              <WithStyles(TableCell)
                align="left"
                className="draws"
              />
              <WithStyles(TableCell)
                align="left"
                className="isturn"
              />
            </WithStyles(TableRow)>
          </WithStyles(TableBody)>,
        ],
        "id": "rpc-table",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(TableRow)>
              <WithStyles(TableCell)
                align="left"
                className="ngames"
              >
                GAMES
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="wins"
              >
                WINS
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="losses"
              >
                LOSSES
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="draws"
              >
                DRAWS
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="turn"
              >
                TURN
              </WithStyles(TableCell)>
            </WithStyles(TableRow)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(TableCell)
                  align="left"
                  className="ngames"
                >
                  GAMES
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)
                  align="left"
                  className="wins"
                >
                  WINS
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)
                  align="left"
                  className="losses"
                >
                  LOSSES
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)
                  align="left"
                  className="draws"
                >
                  DRAWS
                </WithStyles(TableCell)>,
                <WithStyles(TableCell)
                  align="left"
                  className="turn"
                >
                  TURN
                </WithStyles(TableCell)>,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": "GAMES",
                  "className": "ngames",
                },
                "ref": null,
                "rendered": "GAMES",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": "WINS",
                  "className": "wins",
                },
                "ref": null,
                "rendered": "WINS",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": "LOSSES",
                  "className": "losses",
                },
                "ref": null,
                "rendered": "LOSSES",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": "DRAWS",
                  "className": "draws",
                },
                "ref": null,
                "rendered": "DRAWS",
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": "TURN",
                  "className": "turn",
                },
                "ref": null,
                "rendered": "TURN",
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <WithStyles(TableRow)
              className="score-row"
            >
              <WithStyles(TableCell)
                align="left"
                className="games"
              />
              <WithStyles(TableCell)
                align="left"
                className="wins"
              />
              <WithStyles(TableCell)
                align="left"
                className="losses"
              />
              <WithStyles(TableCell)
                align="left"
                className="draws"
              />
              <WithStyles(TableCell)
                align="left"
                className="isturn"
              />
            </WithStyles(TableRow)>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <WithStyles(TableCell)
                  align="left"
                  className="games"
                />,
                <WithStyles(TableCell)
                  align="left"
                  className="wins"
                />,
                <WithStyles(TableCell)
                  align="left"
                  className="losses"
                />,
                <WithStyles(TableCell)
                  align="left"
                  className="draws"
                />,
                <WithStyles(TableCell)
                  align="left"
                  className="isturn"
                />,
              ],
              "className": "score-row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": undefined,
                  "className": "games",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": undefined,
                  "className": "wins",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": undefined,
                  "className": "losses",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": undefined,
                  "className": "draws",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "align": "left",
                  "children": undefined,
                  "className": "isturn",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": <WithStyles(Table)
          id="rpc-table"
        >
          <WithStyles(TableHead)>
            <WithStyles(TableRow)>
              <WithStyles(TableCell)
                align="left"
                className="ngames"
              >
                GAMES
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="wins"
              >
                WINS
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="losses"
              >
                LOSSES
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="draws"
              >
                DRAWS
              </WithStyles(TableCell)>
              <WithStyles(TableCell)
                align="left"
                className="turn"
              >
                TURN
              </WithStyles(TableCell)>
            </WithStyles(TableRow)>
          </WithStyles(TableHead)>
          <WithStyles(TableBody)>
            <WithStyles(TableRow)
              className="score-row"
            >
              <WithStyles(TableCell)
                align="left"
                className="games"
              />
              <WithStyles(TableCell)
                align="left"
                className="wins"
              />
              <WithStyles(TableCell)
                align="left"
                className="losses"
              />
              <WithStyles(TableCell)
                align="left"
                className="draws"
              />
              <WithStyles(TableCell)
                align="left"
                className="isturn"
              />
            </WithStyles(TableRow)>
          </WithStyles(TableBody)>
        </WithStyles(Table)>,
        "className": "paper-root",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <WithStyles(TableHead)>
              <WithStyles(TableRow)>
                <WithStyles(TableCell)
                  align="left"
                  className="ngames"
                >
                  GAMES
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="wins"
                >
                  WINS
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="losses"
                >
                  LOSSES
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="draws"
                >
                  DRAWS
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="turn"
                >
                  TURN
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>
            </WithStyles(TableHead)>,
            <WithStyles(TableBody)>
              <WithStyles(TableRow)
                className="score-row"
              >
                <WithStyles(TableCell)
                  align="left"
                  className="games"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="wins"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="losses"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="draws"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="isturn"
                />
              </WithStyles(TableRow)>
            </WithStyles(TableBody)>,
          ],
          "id": "rpc-table",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableRow)>
                <WithStyles(TableCell)
                  align="left"
                  className="ngames"
                >
                  GAMES
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="wins"
                >
                  WINS
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="losses"
                >
                  LOSSES
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="draws"
                >
                  DRAWS
                </WithStyles(TableCell)>
                <WithStyles(TableCell)
                  align="left"
                  className="turn"
                >
                  TURN
                </WithStyles(TableCell)>
              </WithStyles(TableRow)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(TableCell)
                    align="left"
                    className="ngames"
                  >
                    GAMES
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell)
                    align="left"
                    className="wins"
                  >
                    WINS
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell)
                    align="left"
                    className="losses"
                  >
                    LOSSES
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell)
                    align="left"
                    className="draws"
                  >
                    DRAWS
                  </WithStyles(TableCell)>,
                  <WithStyles(TableCell)
                    align="left"
                    className="turn"
                  >
                    TURN
                  </WithStyles(TableCell)>,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": "GAMES",
                    "className": "ngames",
                  },
                  "ref": null,
                  "rendered": "GAMES",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": "WINS",
                    "className": "wins",
                  },
                  "ref": null,
                  "rendered": "WINS",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": "LOSSES",
                    "className": "losses",
                  },
                  "ref": null,
                  "rendered": "LOSSES",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": "DRAWS",
                    "className": "draws",
                  },
                  "ref": null,
                  "rendered": "DRAWS",
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": "TURN",
                    "className": "turn",
                  },
                  "ref": null,
                  "rendered": "TURN",
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <WithStyles(TableRow)
                className="score-row"
              >
                <WithStyles(TableCell)
                  align="left"
                  className="games"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="wins"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="losses"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="draws"
                />
                <WithStyles(TableCell)
                  align="left"
                  className="isturn"
                />
              </WithStyles(TableRow)>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <WithStyles(TableCell)
                    align="left"
                    className="games"
                  />,
                  <WithStyles(TableCell)
                    align="left"
                    className="wins"
                  />,
                  <WithStyles(TableCell)
                    align="left"
                    className="losses"
                  />,
                  <WithStyles(TableCell)
                    align="left"
                    className="draws"
                  />,
                  <WithStyles(TableCell)
                    align="left"
                    className="isturn"
                  />,
                ],
                "className": "score-row",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": undefined,
                    "className": "games",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": undefined,
                    "className": "wins",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": undefined,
                    "className": "losses",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": undefined,
                    "className": "draws",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "align": "left",
                    "children": undefined,
                    "className": "isturn",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
